scripts/generate_conda_file.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script creates yaml files to build conda environments
+# For generating a conda file for running only python code:
+# $ python generate_conda_file.py
+# For generating a conda file for running python gpu:
+# $ python generate_conda_file.py --gpu
+# For generating a conda file for running pyspark:
+# $ python generate_conda_file.py --pyspark
+# For generating a conda file for running python gpu and pyspark:
+# $ python generate_conda_file.py --gpu --pyspark
+# For generating a conda file for running python gpu and pyspark with a particular version:
+# $ python generate_conda_file.py --gpu --pyspark-version 2.4.0
+
+import argparse
+
+
+CHANNELS = ["conda-forge", "pytorch", "fastai", "defaults"]
+
+CONDA_BASE = {
+    "dask": "dask>=0.17.1",
+    "fastai": "fastai>=1.0.40",
+    "fastparquet": "fastparquet>=0.1.6",
+    "gitpython": "gitpython>=2.1.8",
+    "ipykernel": "ipykernel>=4.6.1",
+    "jupyter": "jupyter>=1.0.0",
+    "matplotlib": "matplotlib>=2.2.2",
+    "numpy": "numpy>=1.13.3",
+    "pandas": "pandas>=0.23.4",
+    "pymongo": "pymongo>=3.6.1",
+    "python": "python==3.6.8",
+    "pytest": "pytest>=3.6.4",
+    "seaborn": "seaborn>=0.8.1",
+    "scikit-learn": "scikit-learn==0.19.1",
+    "scipy": "scipy>=1.0.0",
+    "scikit-surprise": "scikit-surprise>=1.0.6",
+    "tensorflow": "tensorflow==1.12.0",
+}
+
+CONDA_PYSPARK = {"pyarrow": "pyarrow>=0.8.0", "pyspark": "pyspark==2.3.1"}
+
+CONDA_GPU = {"numba": "numba>=0.38.1", "tensorflow": "tensorflow-gpu==1.12.0"}
+
+PIP_BASE = {
+    "azureml-sdk[notebooks,contrib]": "azureml-sdk[notebooks,contrib]>=1.0.8",
```

Comment:
a couple of suggestions here:
`"azureml-sdk": "azureml-sdk[notebooks,contrib]==1.0.10",` 
I got an error today due to the use of `>=`

Reply:
ah, ok, np